spec/lib/open_project/journal_formatter/project_phase_definition_spec.rb

```diff
@@ -0,0 +1,264 @@
+      let(:project_phase) { build_stubbed(:project_phase, active: false) }
+      let(:other_project_phase) { build_stubbed(:project_phase, active: false) }
+      let(:old_value) { other_project_phase.definition_id.to_s }
+      let(:new_value) { project_phase.definition_id.to_s }
+      let(:expected) do
+        I18n.t(:text_journal_changed_plain,
+               label: "<strong>Project phase</strong>",
+               linebreak: nil,
+               old: "<i>#{other_project_phase.name} (Inactive)</i>",
+               new: "<i>#{project_phase.name} (Inactive)</i>")
+      end
+
+      it_behaves_like "renders project phase definition change"
+    end
+
+    context "when deleting an inactive phase" do
+      let(:project_phase) { build_stubbed(:project_phase, active: false) }
+      let(:old_value) { project_phase.definition_id.to_s }
+      let(:new_value) { nil }
+      let(:expected) do
+        I18n.t(:text_journal_deleted,
+               label: "<strong>Project phase</strong>",
+               old: "<strike><i>#{project_phase.name} (Inactive)</i></strike>")
+      end
+
+      it_behaves_like "renders project phase definition change"
+    end
+
+    context "when setting a phase not configured for the project" do
+      let(:project) { build_stubbed(:project, phases: []) }
+      let(:old_value) { nil }
+      let(:new_value) { project_phase.definition_id.to_s }
+      let(:expected) do
+        I18n.t(:text_journal_set_to,
+               label: "<strong>Project phase</strong>",
+               value: "<i>#{project_phase.name} (Inactive)</i>")
+      end
+
+      it_behaves_like "renders project phase definition change"
+    end
+
+    context "when changing between two phases not configured for the project" do
+      let(:project) { build_stubbed(:project, phases: []) }
+      let(:old_value) { other_project_phase.definition_id.to_s }
+      let(:new_value) { project_phase.definition_id.to_s }
+      let(:expected) do
+        I18n.t(:text_journal_changed_plain,
+               label: "<strong>Project phase</strong>",
+               linebreak: nil,
+               old: "<i>#{other_project_phase.name} (Inactive)</i>",
+               new: "<i>#{project_phase.name} (Inactive)</i>")
+      end
+
+      it_behaves_like "renders project phase definition change"
+    end
+
+    context "when deleting a phase not configured for the project" do
+      let(:project) { build_stubbed(:project, phases: []) }
+      let(:old_value) { project_phase.definition_id.to_s }
+      let(:new_value) { nil }
+      let(:expected) do
+        I18n.t(:text_journal_deleted,
+               label: "<strong>Project phase</strong>",
+               old: "<strike><i>#{project_phase.name} (Inactive)</i></strike>")
+      end
+
+      it_behaves_like "renders project phase definition change"
+    end
+
+    context "when setting a phase whose definition is deleted" do
+      let(:old_value) { nil }
+      let(:new_value) { -1.to_s }
```

Comment:
I'd expect rubocop to annoy about this:
```suggestion
      let(:new_value) { "-1" }
```